### PR TITLE
Improvements to Notification Page

### PR DESCRIPTION
**Pull request by BBScholar**
- improved design of UI on notification page
- regex for checking valid email
- don't allow submission of form until 1 or more classes are select and email is valid
- some formatting
- Changed popup windows

TODO:
- Limit width of boxes to ~90% of window width

### CODE REVIEW

**Review by InvincibleRMC — APPROVED** [review]

File: back_end/server.py [hunk 1]
```diff
@@ -44,46 +51,48 @@ def GetSchedule(self, request, context) -> ScheduleResponse:
...
-        print(f"Received professor request: {repr(request)}")
+        print(f"Received professor request:", request)
         res = ProfessorResponse()
         res.professors.extend([self.create_prof(1, "Ronald", "Loui"),
                                self.create_prof(2, "Harold", "Connamacher")])
         return res
-
-    def GetCourses(self, request, context) -> CourseResponse:
+        # return super().GetProfessors(request, context)
+
```
Comment (BBScholar) [inline back_end/server.py:70]: It's removing type annotations. Merge from main

File: front_end/lib/network.dart [hunk 2]
```diff
@@ -1,17 +1,24 @@
...
 import 'package:front_end/proto/requests.pb.dart';
 import 'package:front_end/proto/responses.pb.dart';
 
 import 'package:grpc/grpc.dart' as grpc;
 
+import 'package:front_end/proto/service.pb.dart';
+import 'package:front_end/proto/service.pbserver.dart';
+
 class Network {
-  static const String _host = "localhost";
```
Comment (BBScholar) [inline front_end/lib/network.dart:14]: Not good.

File: front_end/lib/network.dart [hunk 3]
```diff
@@ -20,7 +27,7 @@ class Network {
     return _instance;
   }
 
-  GrpcOrGrpcWebClientChannel? _channel;
```
Comment (BBScholar) [inline front_end/lib/network.dart:23]: This is really not good.

**Review by BBScholar — COMMENTED** [review]
Looks good to me. Works on my laptop, both web and desktop.

**Pull request merged**